src/base/rss/rss_session.cpp

```diff
@@ -0,0 +1,489 @@
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#include "rss_session.h"
+
+#include <QDebug>
+#include <QDir>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QJsonValue>
+#include <QSaveFile>
+#include <QString>
+#include <QThread>
+#include <QVariantHash>
+
+#include "../asyncfilestorage.h"
+#include "../logger.h"
+#include "../profile.h"
+#include "../settingsstorage.h"
+#include "../utils/fs.h"
+#include "rss_article.h"
+#include "rss_feed.h"
+#include "rss_item.h"
+#include "rss_folder.h"
+
+const int MsecsPerMin = 60000;
+const QString ConfFolder(QStringLiteral("rss/config"));
+const QString DataFolderName(QStringLiteral("rss/data"));
+const QString ItemsFile(QStringLiteral("items.json"));
+
+const QString SettingsKey_ProcessingEnabled(QStringLiteral("RSS/Session/EnableProcessing"));
+const QString SettingsKey_RefreshInterval(QStringLiteral("RSS/Session/RefreshInterval"));
+const QString SettingsKey_MaxArticlesPerFeed(QStringLiteral("RSS/Session/MaxArticlesPerFeed"));
+
+using namespace RSS;
+
+QPointer<Session> Session::m_instance = nullptr;
+
+Session::Session()
+    : m_workingThread(new QThread(this))
+    , m_processingEnabled(SettingsStorage::instance()->loadValue(SettingsKey_ProcessingEnabled, false).toBool())
+    , m_refreshInterval(SettingsStorage::instance()->loadValue(SettingsKey_RefreshInterval, 30).toUInt())
+    , m_maxArticlesPerFeed(SettingsStorage::instance()->loadValue(SettingsKey_MaxArticlesPerFeed, 50).toInt())
+{
+    Q_ASSERT(!m_instance); // only one instance is allowed
+    m_instance = this;
+
+    m_confFileStorage = AsyncFileStorage::create(
+                Utils::Fs::expandPathAbs(specialFolderLocation(SpecialFolder::Data) + ConfFolder));
```

Comment:
I think you should use the `SpecialFolder::Config` for this and make `ConfFolder = "rss"` and `ItemsFile = "feeds.json"`. I also think you should place the download rules in the same location. Only the articles should in the data folder.

Speaking of the config storage overall, I think you should change some of the folder names as well. Here is what I think makes the most sense:
`CONFIG_PATH/rss/feeds.json`
`CONFIG_PATH/rss/download_rules.json`
`DATA_PATH/rss/articles/NORMALIZED_FEED_NAME.json`